keel-core/src/main/kotlin/com/netflix/spinnaker/keel/api/NotificationConfig.kt

```diff
@@ -0,0 +1,39 @@
+/*
+ *
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package com.netflix.spinnaker.keel.api
+
+// todo eb: this does not allow you to customize the notification message, but we can add that later.
+data class NotificationConfig(
+  val type: NotificationType,
+  val address: String, // either slack channel or email address
+  val frequency: NotificationFrequency
+)
+
+enum class NotificationFrequency {
```

Comment:
I don't know if this is the right strategy, but I do know that I don't want users to have to configure notifications for "when a pipeline starts" because I want to hide these things from them.

Maybe it's better to just have a list of events that you can notify on, like `ACUTATION_STARTING`, `ACUTATION_COMPLETE`, `ACUTATION_FAILED`? However, if we call out explicit states, we can never add more or less. I sort of like the idea of just specifying the level and then figuring out what fits in there... 

Thoughts?

Reply:
could add `DEFAULT` to the enum and make frequency in `NotificationConfig` a `Map<NotificationFrequency, Boolean>` as to whether to notify on a particular state (maybe it should be renamed to `NotificationStatus` or something in that case)

so I could have:
```
frequency:
  DEFAULT: true
  ACTUATION_STARTING: false
```

And I would get notified on complete/failed. And I'd also get notified on any future state we added until I explicitly opted out of it.

Actually come to think of it maybe the default should always be true, and we have a list of states to explicitly suppress..?

Reply:
I don't really have much of an opinion on the `DEFAULT` thing, or map vs. not map. Feels like the need for specifying exact types of notification events will come up later, but could easily see adding another field (parallel to `frequency`) that would let you turn on/off event types.

Reply:
I'm not sure I understand why it means we can't add things in the future. We can always run migrations or have defaults for event types we add in future. I'm concerned that describing it in the sense of levels might be too obscure for people or we'll get bugged for "I want NORMAL, except…". But I think this is a great start.

Reply:
I just mean not easily - Like, if you have to specify "starting"/"complete"/"failed"/etc events it's hard to keep everyone in "recommenced" settings. 

I think, let's tackle this in the future with maybe something like `overrides` as a parallel field to frequency, where someone could add or subtract notifications to their hearts content.